portfolio/src/main/webapp/gallery_script.js

```diff
@@ -14,3 +14,7 @@ function createListElement(text) {
   liElement.innerText = text;
   return liElement;
 }
+
+function deleteComments() {
+  fetch('/delete-data', {method: 'POST'}).then(fetchComments());
```

Comment:
then() takes a function reference which it calls once the promise (in this case, the fetch result) resolves. I'd expect `fetchComments` to be the function reference passed in, but instead `fetchComments()` (note the parenthesis which call the function) is passed in - meaning `fetchComments()` is being called at the same time as `fetch` instead of after it is done. This results in a race between the deletion request and the fetch request, where the comments will seemingly randomly remain/disappear in the end.

All this is to say, it'd probably be better to delete the parenthesis after `fetchComments` so that the behavior is more predictable and correct in all cases

Reply:
I didn't clear this up before so might as well now: because `fetchComments()` is called right away, it's return value is actually what is being passed to `then()`, but AFAIK it has no return value which means nothing happens later. So the current implementation is the same as:

    fetchComments();
    fetch('/delete-data', {method: 'POST'});